packages/platform-sdk-profiles/src/profiles/profile.ts

```diff
@@ -381,8 +383,8 @@ export class Profile implements ProfileContract {
 	private decrypt(password: string): ProfileStruct {
 		const usesPassword: boolean = this.#data.password !== undefined;
 
-		if (usesPassword && password === undefined) {
-			throw new Error("This profile uses a password but none was passed for decryption.");
+		if (!usesPassword) {
+			throw new Error("This profile does not use a password but password was passed for decryption");
```

Comment:
@faustbrian this condition was unreachable because `restore` ensures password is always defined. Modified the condition to throw when password is provided for a non-password profile restore.